developers/advanced/null-value-support.md

```diff
@@ -1,110 +1,184 @@
 # Null value support
 
 {% hint style="danger" %}
-**Multi-stage engine warning**
+**Version warning**
 
-This document describes null handling for the [single-stage query engine](../../reference/single-stage-engine.md). At this time, the [multi-stage query engine](../../reference/multi-stage-engine.md) (v2) does not support null handling. Queries involving null values in a multi-stage environment may return unexpected results.
+This document describes null handling for Apache Pinot 1.0.0.
+This version includes two important limitations that have been addressed in the next (yet to release) version:
+1. Null handling is not supported in the [multi-stage query engine](../../reference/multi-stage-engine.md).
+2. At ingestion time, null handling is a table-level configuration and does not support column-level configuration.
 {% endhint %}
 
-Null handling is defined in two different parts: at ingestion and at query time.
-* [Basic null handling support](#basic-null-handling-support) means that you have enabled null handling at ingestion.
-* [Advanced null support](#advanced-null-handling-support) means that you have also enabled null handling at query time.
-
-## Basic null handling support
-
-By default, null handling is disabled (`nullHandlingEnabled=false`) in the Table index configuration ([tableIndexConfig](https://docs.pinot.apache.org/configuration-reference/table#tableindexconfig-1)). When null support is disabled, `IS NOT NULL` evaluates to `true,` and `IS NULL` evaluates to `false`. For example, the predicate in the query below matches all records.
+For performance reasons, null handling support is disabled by default in Apache Pinot.
+When null support is disabled, all columns are treated as not null.
+Predicates like `IS NOT NULL` evaluates to `true,` and `IS NULL` evaluates to `false`.
+Aggregation functions like `COUNT`, `SUM`, `AVG`, `MODE`, etc. treat all columns as not null.
 
+For example, the predicate in the query below matches all records.
 <pre class="language-sql"><code class="lang-sql"><strong>select count(*) from my_table where column IS NOT NULL
 </strong></code></pre>
 
-### Enable basic null support
-
-To enable basic null support (`IS NULL` and `IS NOT NULL`) and generate the null index, in the Table index configuration ([tableIndexConfig](https://docs.pinot.apache.org/configuration-reference/table#tableindexconfig-1)), set `nullHandlingEnabled=true`.
-
-When null support is enabled, `IS NOT NULL` and `IS NULL` evaluate to `true` or `false` according to whether a null is detected.
-
-{% hint style="info" %}
-**Important**
+In order to handle null values in your data, you must first indicate Pinot to store nulls in your tables.
```

Comment:
To handle null values in your data, you must specify nulls can be store in Pinot tables.

is this statement accurate, @gortiz? 

if so, where do you specify this nulls can be handled--is this what's discuss below?
regarding `enableNullHandling`?

Reply:
> is this statement accurate, @gortiz?

Yes

> if so, where do you specify this nulls can be handled--is this what's discuss below? regarding enableNullHandling?

Yep. Maybe we rephrase this part as:

`To handle null values in your data, you must configure your tables to store nulls at ingestion time`

It may be easier to understand given the later section is called `Storing nulls at ingestion time`

What do you think?

Reply:
yes, thank you, @gortiz!